include/openthread/channel_monitor.h

```diff
@@ -62,6 +62,45 @@ extern "C" {
  *
  */
 
+/**
+ * This function starts the Channel Monitoring operation.
+ *
+ * Once stated, any previously collected data is cleared.
+ *
+ * @note OpenThread core internally starts the Channel Monitoring operation when the IPv6 interface is brought up (call
+ * to `otIp6SetEnabled()` to enable the interface).
+ *
+ * @param[in]  aInstance       A pointer to an OpenThread instance.
+ *
+ * @retval OT_ERROR_NONE      Channel Monitoring started successfully.
+ * @retval OT_ERROR_ALREADY   Channel Monitoring has already been started.
+ *
+ */
+otError otChannelMonitorStart(otInstance *aInstance);
```

Comment:
In the past, @pvanhorn made the argument that Start/Stop APIs should be collapsed into a single function with a bool argument.  Thoughts?

Reply:
Wow, that kid has a good memory <impressed>.  I have to agree with Jonathan and my younger self that such an API would be preferable.

Reply:
Changed the public API name to `otChannelMonitorSetEnabled`... but keeping core internal method as is... :)

I slightly prefer `Start/Stop` for the core code... but I agree for public OT API and for NCP/spinel interaction, `SetEnabled` would be preferable as it can be mapped directly to a Boolean property.